esp32_s3_korvo_2/include/bsp/esp32_s3_korvo_2.h

```diff
@@ -99,11 +98,50 @@ typedef enum {
     BSP_BUTTON_SET,
     BSP_BUTTON_VOLDOWN,
     BSP_BUTTON_VOLUP,
-    BSP_BUTTON_NUM,
     BSP_BUTTON_MAIN,
+    BSP_BUTTON_NUM,
 } bsp_button_t;
 
-extern const button_config_t bsp_button_config[BSP_BUTTON_NUM];
+/**
+ * @brief Set button's GPIO as input
+ *
+ * @param[in] btn Button to be initialized
+ * @return
+ *     - ESP_OK Success
+ */
+esp_err_t bsp_button_init(const bsp_button_t btn)
+__attribute__((deprecated("use bsp_iot_button_create instead")));
+
+/**
+ * @brief Get button's state
+ *
+ * Note: For LCD panel button which is defined as BSP_BUTTON_MAIN, bsp_display_start should
+ *       be called before call this function.
+ *
+ * @param[in] btn Button to read
+ * @return true  Button pressed
+ * @return false Button released
+ */
+bool bsp_button_get(const bsp_button_t btn)
+__attribute__((deprecated("use espressif/button API instead")));
+
+/**
+ * @brief Initialize all buttons
+ *
+ * Returned button handlers must be used with espressif/button component API
+ *
+ * @note For LCD panel button which is defined as BSP_BUTTON_MAIN, bsp_display_start should
+ *       be called before call this function.
+ *
+ * @param[out] btn_array      Output button array
+ * @param[out] btn_cnt        Number of button handlers saved to btn_array, can be NULL
+ * @param[in]  btn_array_size Size of output button array. Must be at least BSP_BUTTON_NUM
+ * @return
+ *     - ESP_OK               All buttons initialized
+ *     - ESP_ERR_INVALID_ARG  btn_array is too small or NULL
+ *     - ESP_FAIL             Underlaying iot_button_create failed
+ */
+esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int btn_array_size);
```

Comment:
Is it good to initialize all buttons? Cannot be situation, when button will be shared with something else and we cannot initialize button. For example USB/JTAG or other peripherals...

Reply:
Agree. Closing conversation.